knowledge-base/how-to-fix-axe-errors-for-html5-viewer.md

```diff
@@ -0,0 +1,110 @@
+[Axe web accessibility testing tool](https://www.deque.com/axe/) might report violations and warnings releated to HTML5 Report Viewer and the content of the report.
+
+When providing accessibility for the report viewer we were guided by [Section 508 of the Rehabilitation Act](http://www.section508.gov/), 
+[Web Content Accessibility Guidelines (WCAG) 2.0](https://www.w3.org/TR/WCAG20/), and the standards set by [WAI-ARIA](https://www.w3.org/TR/wai-aria-practices/). 
+The implemented approach was tested using [Microsoft Narrator](https://en.wikipedia.org/wiki/Microsoft_Narrator) and [JAWS](https://www.freedomscientific.com/products/software/jaws/)
+screen readers.
+
+As there is no common stardard for implementing the accessibility for web applications some of the accessibility testing tools might 
+have additional guidelinies for implementing accessibility which are not respected by HTML5 Report Viewer.
+
+## Error Messages
+
+Elements should not have tabindex greater than zero
+
+Certain ARIA roles must contain particular children
+
+Elements must have sufficient color contrast
+
+
+## Solution
+
+In Telerik Reporting R3 2017 (version 11.2.17.913) we introduced HTML5 Report Viewer accessibility affecting both the viewer controls 
+and the report content. The accessibility features are enabled or disabled using *enableAccessibility* option of the viewer
+(its default value is false).
+
+As a first step, make sure that accessibility is enabled for the viewer as this step will help to eliminate most of the reported
+warnings and violations:
+
+```JavaScript
+$("#reportViewer1")
+                .telerik_ReportViewer({
+                        enableAccessibility: true,
+```
+
+Below are the solutions to eliminate the remaining violations reported by Axe.
+
+*Elements should not have tabindex greater than zero*
+
+The viewer can comply with this requirement by having all tabindex attributes set to 0. Changing the value of all tabindex attributes 
+can be achieved using *pageReady()* event of the viewer:
+
+```JavaScript
+pageReady: function (e, args) {
+                        $("[tabindex]").attr("tabindex", "0");
+                    }
+```
+
+*Certain ARIA roles must contain particular children*
+
+The viewer renders [Document Map element](../designing-reports-document-map) with data-role="treeview" attribute. When the Document Map 
+has no items the parent element will still be rendered but without the treeitem nodes. This is treated as a violation by the Axe. 
+The solution is to [add an item to the DocumentMap](../designing-reports-interactivity-how-to-add-document-map) so the treeitem node 
+is appended to the treeview parent in the viewer.
+
+The DocumentMap is visible by default on the initial rendering of the viewer. To hide it, you can set *documentMapVisible*
+option of the viewer to *false*:
+
+```JavaScript
+$("#reportViewer1")
+                .telerik_ReportViewer({
+                          documentMapVisible: false
+```
+
```

Comment:
Good solution 👍 
I would only add that the ARIA role of the document map is set in the viewer template (line 174: role="navigation"), so the users can edit the template and remove the role if necessary.